packages/create-audius-app/helpers/is-online.ts

```diff
@@ -0,0 +1,45 @@
+import { execSync } from 'child_process'
+import dns from 'dns/promises'
+import url from 'url'
+
+function getProxy(): string | undefined {
+  if (process.env.https_proxy) {
+    return process.env.https_proxy
+  }
+
+  try {
+    const httpsProxy = execSync('npm config get https-proxy').toString().trim()
+    return httpsProxy !== 'null' ? httpsProxy : undefined
+  } catch (e) {
+    return
+  }
+}
+
+export async function getOnline(): Promise<boolean> {
```

Comment:
are these based off existing create-app code out there, cause this is so comprehensive, v cool

Reply:
oh i see now, its based on create-turbo etc